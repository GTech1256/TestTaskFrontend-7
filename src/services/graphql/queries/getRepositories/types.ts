import { LicenseType } from "../getAllLicensies"

/** incomplete description */
type UserType = {
  id: string
  avatarUrl: string
}

export type PageInfoType = {
  endCursor: string
  hasNextPage: boolean
  hasPreviousPage: boolean
  startCursor: string
}

type StargazerConnectionType = {
  nodes: UserType[]
  pageInfo: PageInfoType
  totalCount: number
}

type RepositoryOwnerType = {
  id: string
  login: string
  repository?: RepositoryType
  resourcePath: string
  url: string
}

type CodeOfConductType = {
  body?: string
  id: string
  key: string
  name: string
  resourcePath: string
  url: string
}

/** incomplete description */
export type RepositoryType = {
  id: number
  databaseId: number
  description: string
  codeOfConduct: CodeOfConductType
  name: string
  owner: RepositoryOwnerType
  url: string
  stargazers: StargazerConnectionType
  licenseInfo: LicenseType
}
