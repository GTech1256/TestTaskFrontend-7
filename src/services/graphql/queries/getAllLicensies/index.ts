export { ALL_LICENSIES } from './getAllLicensies.query'
export type { QueryType as getAllLicensiesQueryType  } from './getAllLicensies.query'
export type { LicenseType } from './types'