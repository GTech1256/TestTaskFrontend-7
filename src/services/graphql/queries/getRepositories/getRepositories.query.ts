import { gql } from '@apollo/client';

import { RepositoryType, PageInfoType } from './types';


export type RepositoryResponseType = {
  description: RepositoryType['description']
  name: RepositoryType['name']
  url: RepositoryType['url']
  stargazers: {
    totalCount: RepositoryType['stargazers']['totalCount']
  }
  owner: {
    login: RepositoryType['owner']['login']
  }
}

export type QueryType = {
  search: {
    pageInfo: PageInfoType
    edges: {
      node: RepositoryResponseType
    }[]
  }
}

export const ALL_REPOSITORIES = gql`
  query getRepositories($query: String!, $after: String, $before: String) {
    search(type: REPOSITORY, query: $query, first:10, after:$after, before: $before) {
      pageInfo {
        endCursor
        hasNextPage
        hasPreviousPage
        startCursor
      }
      edges {

        node {
          ... on Repository {
            description
            name
            url
            stargazers {
              totalCount
            }
            owner {
              login
            }
          }
        }
      }
    }
  }
`;