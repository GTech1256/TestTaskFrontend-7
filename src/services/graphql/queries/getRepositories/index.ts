export { ALL_REPOSITORIES } from './getRepositories.query'
export { getRepositiryQueryString, MIN_STARS_COUNT } from './utils'
export type { QueryType as AllRepositoriesQueryType, RepositoryResponseType } from './getRepositories.query'