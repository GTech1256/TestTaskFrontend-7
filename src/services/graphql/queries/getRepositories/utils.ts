type RepositiryQueryStringOptionsType = {
  license?: string
  name?: string
}

export const MIN_STARS_COUNT = 1600

export const getRepositiryQueryString = ({
  license,
  name,
} : RepositiryQueryStringOptionsType) => `${name ? `${name} in:name` : ''} ${license ? `license:${license}` : ''} language:javascript stars:>${MIN_STARS_COUNT}`