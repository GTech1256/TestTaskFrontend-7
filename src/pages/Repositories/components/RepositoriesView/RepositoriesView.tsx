import React from 'react';

import { AllRepositoriesQueryType } from '../../../../services/graphql/queries/getRepositories';
import RepositoryView from '../RepositoryView';


type Props = {
  className?: string
  reposetories: AllRepositoriesQueryType['search']['edges']
}

const RepositoriesView = ({
  className = '',
  reposetories
}: Props) => (
  <div className={`list-group ${className}`}>
    {reposetories.map(({ node: reposetory }) => <RepositoryView key={reposetory.url} {...reposetory} />)}
  </div>
)

export default RepositoriesView;