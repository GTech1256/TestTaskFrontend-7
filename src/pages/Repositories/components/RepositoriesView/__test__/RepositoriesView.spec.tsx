import React from 'react';
import renderer from 'react-test-renderer';

import RepositoriesView from '../RepositoriesView';
import { RepositoryResponseType } from '../../../../../services/graphql/queries/getRepositories';


const reposetory: { node: RepositoryResponseType } = {
  node: {
    name: 'name',
    owner: {
      login: 'login'
    },
    url: 'url',
    stargazers: { totalCount: 999 },
    description: 'description'
  }
};

describe('<RepositoriesView />', () => {
  it('should render correct', () => {
    const props = {
      reposetories: [
        reposetory
      ]
    };

    const tree = renderer
      .create(<RepositoriesView {...props} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  })
})