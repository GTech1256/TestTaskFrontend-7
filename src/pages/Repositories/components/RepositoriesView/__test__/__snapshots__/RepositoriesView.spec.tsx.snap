// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoriesView /> should render correct 1`] = `
<div
  className="list-group "
>
  <a
    className="list-group-item list-group-item-action"
    href="url"
  >
    <div
      className="d-flex w-100 justify-content-between"
    >
      <h5
        className="mb-1"
      >
        name
      </h5>
      <span
        className="badge badge-text-center badge-dark"
      >
        999
      </span>
    </div>
    <p
      className="mb-1"
    >
      description
    </p>
    <small>
      login
    </small>
  </a>
</div>
`;
